\section{Introduction}
\label{sec:introduction}


Biological sequence alignment search algorithms are widely used in Bioinformatics and Computational biology(BCB), where the match of sequences of database and query is searched[Server Kasap]. The main reason in using such search algorithms is to identify the degree of similarity of a newly discovered biological sequence with the already know sequences[[Server Kasap gapped]. For instance, it can be exploited to determine early disease diagnosis, where biological information of a newly discovered sequence can be obtained by finding the most similar disease gene in the database[Xinyu Guo].


Basic Local Alignment Search Tool(BLAST) is one of the widely used heuristic methodologies, which delivers the best local alignment for even large size of data sets. Although, BLAST has been proved to meet the performance and search sensitivity criteria, the improvements in DNA sequencing technologies, which imply the number of genomic sequences is doubling almost every year, display that even algorithms like BLAST cannot tackle the requirements[TimeLogic]. 


Due to the complexity of sequence alignment when dealing with the huge amount of data, currently used platforms cannot satisfy the computational requirements[Masato Yoshimi,]. The usage of Field Programmable Gate Arrays(FPGAs) has been recently proposed as an effective solution to meet the performance requirements. 


Biological sequence alignment is the major task in the bioinformatics, where newly discovered genome sequence is compared with whole database in order to get information about it. It is performed by aligning query sequence to each subject sequence in a database and comparing them to find matchings \cite{guo2012systolic}.
\\
Although there are several algorithms developed (e.g. Needleman-Wunsch, Smith-Waterman), this paper concentrates on Basic Local Alignment Search Tool (BLAST), since it is widely used algorithm in sequence alignment. The reason why it is widely utilize is that the algorithm is much faster compared to dynamic alignments \cite{kasap2008design}. However, the implementation of the tool in a general purposed computers is causing major timing and computational issues with exponentially growing of DNA databases. For instance, it may take several hours to compare hundreds of sequence in a database on general purposed computers. Thus, it may cause significant bottleneck when it comes to database with millions of bio-sequences \cite{sarkar2010hardware}.
\\
In order to deal with timing issue, the algorithm must be implemented on hardware accelerator platforms. Higher computational efficiency can be achieved by using Field Programmable Gate Arrays (FPGAs) since it enables parallel computation and can be reprogrammed. In this paper we propose a design of an architecture for efficient implementation of BLAST algorithm on FPGA and discuss its resource consumption and timing analysis by using results obtained during synthesizing, functional, and timing simulations. 

