\section{Introduction}
\label{sec:introduction}

Biological sequence alignment is the major task in the bioinformatics, where newly discovered genome sequence is compared with whole database in order to get information about it. It is performed by aligning query sequence to each subject sequence in a database and comparing them to find matchings [1].
\\
Although there are several algorithms developed (e.g. Needleman-Wunsch, Smith-Waterman), this paper concentrates on Basic Local Alignment Search Tool (BLAST), since it is widely used algorithm in sequence alignment. The reason why it is widely utilize is that the algorithm is much faster compared to dynamic alignments [2]. However, the implementation of the tool in a general purposed computers is causing major timing and computational issues with exponentially growing of DNA databases. For instance, it may take several hours to compare hundreds of sequence in a database on general purposed computers. Thus, it may cause significant bottleneck when it comes to database with millions of bio-sequences [3].
\\
In order to deal with timing issue, the algorithm must be implemented on hardware accelerator platforms. Higher computational efficiency can be achieved by using Field Programmable Gate Arrays (FPGAs) since it enables parallel computation and can be reprogrammed. In this paper we propose a design of an architecture for efficient implementation of BLAST algorithm on FPGA and discuss its resource consumption and timing analysis by using results obtained during synthesizing, functional, and timing simulations. 

In paper \cite{vipin2019}
